### PR TITLE
[GPartition] Rename the input argument for dispatch

### PR DESCRIPTION
**Pull request by jalencato**
*Issue #, if available:*

*Description of changes:*

We now reverse the meaning of the original ``--do-dispatch`` input argument.

Successfully test with 
```
python3 -m graphstorm.gpartition.dist_partition_graph \
    --input-path ./test-gsprocessing/ \
    --metadata-filename updated_row_counts_metadata.json \
    --output-path ./gspartition-example/ \
    --num-parts 1 \
    --dgl-tool-path ./dgl/tools \
    --partition-algorithm random \
    --ip-config ip_list.txt \
    --ssh-port 2222 \
    --partition-assignment-only
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gpartition/dist_partition_graph.py [hunk 1]
```diff
@@ -189,7 +191,8 @@ def parse_args() -> argparse.Namespace:
     argparser.add_argument("--ip-config", type=str,
                            help=("A file storing a list of IPs, one line for "
                                 "each instance of the partition cluster."))
-    argparser.add_argument("--do-dispatch", action='store_true')
+    argparser.add_argument("--partition-assignment-only", action='store_true',
+                           help="Only generate partition assignment, do not build DGL graph")
```
Comment (classicsong) [inline python/graphstorm/gpartition/dist_partition_graph.py:195]:
```suggestion
                           help="Only generate partition assignments for nodes, the process will not build the partitioned DGL graph")
```

**Comment by thvasilo** [comment]
Marking as breaking change since this changes a user-facing argument.

**Review by thvasilo — APPROVED** [review]
Changes LGTM, this better communicates intent and ensures we do the file creation by default, which is the most common use case.

Do we have integ tests for dist_partition_graph? We should in the future if not.

**Comment by jalencato** [comment]
> Changes LGTM, this better communicates intent and ensures we do the file creation by default, which is the most common use case.
> 
> Do we have integ tests for dist_partition_graph? We should in the future if not.

Let us create an issue and do it in another PR, it should request us to add some more dependencies.

**Pull request merged**